src/pages/groupComparison/groupSelector/GroupSelector.tsx

```diff
@@ -3,24 +3,36 @@ import { observer } from 'mobx-react';
 import { MakeMobxView } from 'shared/components/MobxView';
 import LoadingIndicator from 'shared/components/loadingIndicator/LoadingIndicator';
 import ErrorMessage from 'shared/components/ErrorMessage';
-import GroupComparisonStore from '../GroupComparisonStore';
 import autobind from 'autobind-decorator';
 import GroupSelectorButton from './GroupSelectorButton';
 import GroupSelectorButtonList from './GroupSelectorButtonList';
 import SelectAllDeselectAll from './SelectAllDeselectAll';
-import { OverlapStrategy } from '../../../shared/lib/comparison/ComparisonStore';
+import ComparisonStore, {
+    OverlapStrategy,
+} from '../../../shared/lib/comparison/ComparisonStore';
+import { computed, observable } from 'mobx';
+import { ComparisonGroup } from '../GroupComparisonUtils';
+import CollapsedGroupsButton from './CollapsedGroupsButton';
 
 export interface IGroupSelectorProps {
-    store: GroupComparisonStore;
+    store: ComparisonStore;
+    groupCollapseThreshold?: number;
+    isGroupDeletable: (group: ComparisonGroup) => boolean;
 }
 
 @observer
 export default class GroupSelector extends React.Component<
     IGroupSelectorProps,
     {}
 > {
+    @observable collapsedGroupsShown = false;
     private dragging = false;
 
+    @autobind
+    private toggleCollapsedGroups() {
```

Comment:
action

Reply:
i know that it's academic because it's only updating one value, but i want to maintain the rule that whenever update state you do it in action because it's a gotcha for other devs